mmv1/third_party/terraform/services/apigee/resource_apigee_api.go

```diff
@@ -0,0 +1,403 @@
+}
+
+func resourceApigeeApiUpdate(d *schema.ResourceData, meta interface{}) error {
+	//For how API proxy api is implemented, just treat an update as create, when the name is same, it will create a new revision
+	return resourceApigeeApiCreate(d, meta)
+}
+
+func resourceApigeeApiRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*transport_tpg.Config)
+	userAgent, err := tpgresource.GenerateUserAgentString(d, config.UserAgent)
+	if err != nil {
+		return err
+	}
+
+	url, err := tpgresource.ReplaceVars(d, config, "{{ApigeeBasePath}}organizations/{{org_id}}/apis/{{name}}")
+	if err != nil {
+		return err
+	}
+	log.Printf("[DEBUG] API proxy read url is: %s", url)
+
+	billingProject := ""
+
+	// err == nil indicates that the billing_project value was found
+	if bp, err := tpgresource.GetBillingProject(d, config); err == nil {
+		billingProject = bp
+	}
+	log.Printf("[DEBUG] resourceApigeeApiRead sendRequest")
+	log.Printf("[DEBUG] resourceApigeeApiRead, url=, 	%s", url)
+	res, err := transport_tpg.SendRequest(transport_tpg.SendRequestOptions{
+		Config:    config,
+		Method:    "GET",
+		Project:   billingProject,
+		RawURL:    url,
+		UserAgent: userAgent,
+	})
+	if err != nil {
+		return transport_tpg.HandleNotFoundError(err, d, fmt.Sprintf("ApigeeApi %q", d.Id()))
+	}
+	log.Printf("[DEBUG] resourceApigeeApuRead sendRequest completed")
+	previousLastModifiedAt := getApigeeApiLastModifiedAt(d)
+	if err := d.Set("meta_data", flattenApigeeApiMetaData(res["metaData"], d, config)); err != nil {
+		return fmt.Errorf("Error reading API proxy: %s", err)
+	}
+	currentLastModifiedAt := getApigeeApiLastModifiedAt(d)
+	if err := d.Set("name", flattenApigeeApiName(res["name"], d, config)); err != nil {
+		return fmt.Errorf("Error reading API proxy: %s", err)
+	}
+	if err := d.Set("revision", flattenApigeeApiRevision(res["revision"], d, config)); err != nil {
+		return fmt.Errorf("Error reading API proxy: %s", err)
+	}
+	if err := d.Set("latest_revision_id", flattenApigeeApiLatestRevisionId(res["latestRevisionId"], d, config)); err != nil {
+		return fmt.Errorf("Error reading API proxy: %s", err)
+	}
+
+	//setting hash to suggest update
+	if previousLastModifiedAt != currentLastModifiedAt {
+		d.Set("md5hash", "UNKNOWN")
+		d.Set("detect_md5hash", "UNKNOWN")
+	}
+	return nil
+}
+
+func getApigeeApiLastModifiedAt(d *schema.ResourceData) string {
+
+	metaDataRaw := d.Get("meta_data").([]interface{})
+	if len(metaDataRaw) != 1 {
+		//in Terraform Schema, a nest in object is implemented as an array of length one, even if it's technically an object
+		return "UNKNOWN"
+	}
+	metaData := metaDataRaw[0].(map[string]interface{})
+	if metaData == nil {
+		return "UNKNOWN"
+	}
+	lastModifiedAt := metaData["last_modified_at"].(string)
+	if lastModifiedAt == "" {
+		return "UNKNOWN"
+	}
+	return lastModifiedAt
+}
+
+func resourceApigeeApiDelete(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG] resourceApigeeApiDelete")
```

Comment:
We can cleanup some of these lines.

Reply:
Are you referring to the DEBUG log statements? Do you want to remove all of the in the resource?

Reply:
Not all of them, but the logging of where we are in the method can be removed. I would keep one or two like "finished creating resource with ID: xxx"